go-controller/pkg/allocator/udn/allocator.go

```diff
@@ -28,6 +29,39 @@ func (a *Allocator) AllocateVRFTable(networkName string) (uint, error) {
 	return a.calculateIDsFromNetwork("vrf-table", networkName, config.Gateway.UserDefinedNetworkVRFTableBase, config.Default.MaxUserDefinedNetworks, 1)
 }
 
+func (a *Allocator) AllocateV4MasqueradeIPs(networkName string) ([]net.IP, error) {
+	return a.allocateMasqueradeIPs(networkName, "v4", config.Gateway.V4MasqueradeSubnet)
+}
+
+func (a *Allocator) AllocateV6MasqueradeIPs(networkName string) ([]net.IP, error) {
+	return a.allocateMasqueradeIPs(networkName, "v6", config.Gateway.V6MasqueradeSubnet)
+}
+
+func (a *Allocator) allocateMasqueradeIPs(networkName, idPrefix, masqueradeSubnet string) ([]net.IP, error) {
+	numberOfIPs := uint(2)
+	firstID, err := a.calculateIDsFromNetwork(idPrefix+"-masquerade-subnets", networkName, config.Gateway.UserDefinedNetworkMasqueradeIPBase, config.Default.MaxUserDefinedNetworks*numberOfIPs, numberOfIPs)
```

Comment:
I think we should be doing this logic more like how we give out join ips and transit ips using `newIPGenerator` stuff in CM? that's what I had in mind.. given we need this IP in both ovnkube-controller and ovnkube-node...but currently what we do is fine as well given its a util that can be called either from GR bits or from lgw mp0 bits...

Reply:
Ahh didn't know about the `ipGenerator` thing also didn't know about `AddIPOffset` either, I will refactor this PR with those.